### PR TITLE
Python model to dict fix

### LINKED ISSUE #8651: [BUG][Python] model_to_dict fails when x is a dictionary in model_utils.py
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

```python
            if isinstance(value, list):
                if not value or isinstance(value[0], PRIMITIVE_TYPES):
                    # empty list or primitive types
                    result[attr] = value
                elif isinstance(value[0], ModelSimple):
                    result[attr] = [x.value for x in value]
                else:
                    result[attr] = [model_to_dict(x, serialize=serialize) for x in value]
```
In the `else` part, if x is a dictionary, meaning `value` is a list of dictionaries, `model_to_dict` fails. Because x is not a valid and `model_to_dict`, assuming it to be a model, tries to access the `_composed_schemas`.

##### openapi-generator version

master build and released v5.0.1

##### Suggest a fix

Check if x is a model by checking if it has the `_data_store` property. This was the approach before the 5th Jan commit to this file.
https://github.com/OpenAPITools/openapi-generator/pull/8387

### PR DESCRIPTION
**Pull request by code-lucidal58**
Fix #8651 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @tomplus (2018/10) @Jyhess (2019/01) @arun-nalla (2019/11) @spacether (2019/11)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/python/model_utils.mustache [hunk 1]
```diff
@@ -1217,7 +1217,7 @@ def model_to_dict(model_instance, serialize=True):
                 elif isinstance(value[0], ModelSimple):
                     result[attr] = [x.value for x in value]
                 else:
-                    result[attr] = [model_to_dict(x, serialize=serialize) for x in value]
+                    result[attr] = [model_to_dict(x, serialize=serialize) if hasattr(x, '_data_store') else x for x in value]
```
Comment (spacether) [inline modules/openapi-generator/src/main/resources/python/model_utils.mustache:1220]: How about adding a test that will fail if this is broken? If that had existed then this bug would not have been introduced. What do you think?

Comment (code-lucidal58) [reply modules/openapi-generator/src/main/resources/python/model_utils.mustache:1220]: Sure. Will add it

**Comment by spacether** [comment]
@code-lucidal58 I would like to help finish this PR by adding a schema in the python sample spec and then adding a test of it in the sample python client. Can you add me as a colaborator on your repo? It's not letting me push to your branch?
Or do you prefer that I clone your branch into my repo and continue work in a separate new PR?

**Comment by code-lucidal58** [comment]
@spacether It would be better if you clone my branch to your repo. This PR is very old now. Plus I won't be able to add you as a collaborator to CiscoM31 repo (the source repo in this PR) as it is Cisco's GitHub account.

**Comment by code-lucidal58** [comment]
Closing this PR as the issue is resolved here https://github.com/OpenAPITools/openapi-generator/pull/9153

**Pull request closed without merging**